modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1553 @@
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        //Assertions
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("userName"), name);
+        Assert.assertEquals(obj.getString("id"), id);
+
+        String returnedURI = outputScimResponse.getHeaderParamMap().get("Location");
+        String expectedURI = endpoint + "/" + obj.getString("id");
+        Assert.assertEquals(returnedURI, expectedURI);
+
+        if (attributes == null & excludeAttributes == null) {
+
+            Assert.assertTrue(obj.has("emails"));
+
+        } else if (attributes == null & excludeAttributes != null) {
+
+            Assert.assertFalse(obj.has("emails"));
+
+        }
+
+    }
+
+    @DataProvider(name = "dataForGetNotFoundException")
+    public Object[][] dataToGetNotFoundException() {
+
+        return new Object[][]{
+                {"Obama", "userName", null, notFound}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetNotFoundException")
+    public void testGetUserNotFoundException(String name, String attributes, String excludeAttributes,
+                                             int expectedScimResponseStatus)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(endpoint);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForGetCharonException")
+    public Object[][] dataToGetCharonException() {
+
+        return new Object[][]{
+                {"Obama", "userName", null, charon}
```

Comment:
Do we need this data provider?
If it's just one row of data it can be simply moved with the test.

Reply:
Kept the room for more combination if required, By thinking hard cording the test values inside the test case itself is bad practice.